src/monitorService.ts

```diff
@@ -57,15 +58,18 @@ export class TxMonitorService {
     const transactions = await this.wallet.walletStorage.getTransactions(
       await this.wallet.getAddress()
     );
-    for(const transaction of transactions) {
+    const transactionCount = await this.wallet.getTransactionCount();
+    for (const transaction of transactions) {
       const transactionInfo = await this.wallet.provider.getTransaction(transaction.hash);
 
-      if(transactionIsConfirmed(transactionInfo, this.options.neededConfirmations)) {
+      //delete confirmed and out of date transactions
+      if (transactionIsConfirmed(transactionInfo, this.options.neededConfirmations)
+        || (!transactionInfo && transaction.nonce < transactionCount - 1)) {
```

Comment:
What is the point of resend logic if you delete them right after another one is broadcasted?

Reply:
These tx cant be executed since tx with their nonce is already confirmed